core/utils/toolbox.js

```diff
@@ -126,15 +126,16 @@ Blockly.utils.toolbox.toolboxXmlToJson_ = function(toolboxDef) {
       continue;
     }
     var obj = {};
-    obj['kind'] = child.tagName.toUpperCase();
+    var tagName = child.tagName.toUpperCase();
+    obj['kind'] = tagName;
+
     // Store the xml for a block
-    if (child.tagName.toUpperCase() == 'BLOCK') {
-      obj['blockxml'] = Blockly.utils.xml.domToText(child);
-    }
-    // Get the contents for a category.
-    if (child.tagName.toUpperCase() == 'CATEGORY') {
-      obj['contents'] = Blockly.utils.toolbox.toolboxXmlToJson_(child);
-    }
+    (tagName == 'BLOCK' && (obj['blockxml'] = child));
```

Comment:
Please use an if statement.

Reply:
The reason I am doing it this way is to increase performance. It seems to shave off ~5ms for a toolbox with 20 categories and 100 blocks each.

Reply:
That's very surprising. Do you know why it increases performance?

Reply:
I can see that you've cached the call to tagName, we can still do that, but the comparison is between running: 
```
if (condition) {
  statement
}
```
and 
```
(condition) && (statement)
```
Increasing performance doesn't make sense to me.